File: Bye World/byeworld-frontend/src/model/Company.ts
import { Listing } from "./Listing";
import { Review } from "./Review";

export interface Company {
  id?: number;
  name: string;
  description: string;
  logoUrl?: string;
  email: string;
  vat: string;
  address: string;
  reviewsCount?: number;
  listingsCount?: number;
  avgReview?: number;
  listings?: Listing[];
  reviews?: Review[];
}

File: Bye World/ByeWorld-backend/Models/Company.cs
using System.Text.Json.Serialization;

namespace ByeWorld_backend.Models
{
    public class Company
    {
        public long Id { get; set; }
        public string Name { get; set; }= String.Empty;
        public string Description { get; set; } = String.Empty;
        public string LogoUrl { get; set; } = String.Empty;
        public string Email { get; set; } = String.Empty;
        public string Address{ get; set; } = String.Empty;
        public string VAT { get; set; } = String.Empty;
        public List<Listing> Listings { get; set; } = new List<Listing>();
        public List<Review> Reviews { get; set; } = new List<Review>();
        public User? User { get; set; } = null;
    }
}
